### PR TITLE
Models needed for sofc example

### PR DESCRIPTION
**Pull request by AlexNoring**
## Summary/Motivation:
This PR includes new models and changes that are required for the sofc and rsofc flowsheets that will be pushed to examples-pse.

## Changes proposed in this PR:
1.  Added a new argument to the `get_prop()` function in `natural_gas_PR.py` to allow scaled base units
2.  Added the CPU unit model and tests
3.  Added code to build the sofc rom that calculates sofc voltage and outlet conditions

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by eslickj — DISMISSED** [review]

File: idaes/models_extra/power_generation/unit_models/cpu.py
Comment (dallan-keylogic) [inline idaes/models_extra/power_generation/unit_models/cpu.py:808]: Do all these have to be classes? Couldn't you just write:
```
def vent_pressure_fun(x1, x2, x3, x4, x5, x6):
    return - 0.22395635483747847803966E-013 * x1 + 115831.96800000000803266
```

File: idaes/models_extra/power_generation/unit_models/cpu.py
Comment (dallan-keylogic) [inline idaes/models_extra/power_generation/unit_models/cpu.py:243]: Listing the units in the docstring seems redundant with defining the units on the `Var`

File: idaes/models_extra/power_generation/unit_models/cpu.py
Comment (dallan-keylogic) [inline idaes/models_extra/power_generation/unit_models/cpu.py:76]: These should probably all be private methods.

File: idaes/models_extra/power_generation/unit_models/cpu.py
Comment (dallan-keylogic) [inline idaes/models_extra/power_generation/unit_models/cpu.py:315]: Docstrings that don't add anything to simply reading the function name are pointless.

File: idaes/models_extra/power_generation/unit_models/cpu.py
Comment (dallan-keylogic) [inline idaes/models_extra/power_generation/unit_models/cpu.py:151]: Comments that don't add anything to reading the next line aren't really helpful. Comments should either explain *why* you're doing something, or expand on what you're doing beyond the immediate context.

File: idaes/models_extra/power_generation/unit_models/cpu.py
Comment (dallan-keylogic) [inline idaes/models_extra/power_generation/unit_models/cpu.py:643]: Scaling factors for constraints should generally be set with `iscale.constraint_scaling_transform`. If you set a constraint scaling factor, it scales the Jacobian, but IPOPT still uses the unscaled constraint residual to determine constraint convergence. Transforming the constraint has the effect of scaling both the Jacobian and the constraint residual.

Also, are you setting these constraint scaling factors because all the constraints in this model are well-scaled, or to silence some script that checks whether or not objects are scaled? If it's the latter, adding in these dummy scaling factors defeats the purpose of that script.

File: idaes/models_extra/power_generation/unit_models/cpu.py
Comment (dallan-keylogic) [inline idaes/models_extra/power_generation/unit_models/cpu.py:63]: The long-form `CarbonProcessingUnit` should be used instead of `CPU` when defining a model like this.

**Review by dallan-keylogic — APPROVED** [review]
No major problems, but a lot of minor things that would be good to change before this is merged.

**Review by dallan-keylogic — CHANGES_REQUESTED** [review]
Actually, let's hold up for a moment and discuss things further.

**Review by dallan-keylogic — APPROVED** [review]
LGTM

**Comment by eslickj** [comment]
I'll approve, but you'll have to run black on this thing to get the tests passing.

**Comment by ksbeattie** [comment]
@AlexNoring any eta on getting to this?

**Comment by AlexNoring** [comment]
@ksbeattie I'll have some time to work on it by the end of this week.

**Review by eslickj — APPROVED** [review]

File: idaes/models_extra/power_generation/properties/tests/test_sofc_surrogates.py
Comment (lbianchi-lbl) [inline idaes/models_extra/power_generation/properties/tests/test_sofc_surrogates.py:21]: Since Tensorflow is an optional dependency for IDAES, this test file should be skipped if `tensorflow` is not available, e.g. by adding `pytest.importorskip("tensorflow")` anywhere before importing `idaes.models_extra.power_generation.properties.sofc.sofc_keras_surrogate`.

**Review by lbianchi-lbl — CHANGES_REQUESTED** [review]
I've left a comment on how to address the single remaining failing check in `test_sofc_surrogates.py`.

**Comment by ksbeattie** [comment]
@AlexNoring, will you be able to get @lbianchi-lbl suggestion above done today?  We are due to cut the Aug RC today and would like this PR included, if possible.

File: idaes/models_extra/power_generation/properties/tests/test_sofc_surrogates.py
Comment (lbianchi-lbl) [inline idaes/models_extra/power_generation/properties/tests/test_sofc_surrogates.py:21]:
```suggestion

pytest.importorskip("tensorflow", reason="tensorflow must be installed to enable SOFC keras surrogates tests")
```

**Review by lbianchi-lbl — APPROVED** [review]

**Pull request merged**
